src/components/Trade/TradeConfirm/TradeConfirm.tsx

```diff
@@ -118,6 +128,43 @@ export const TradeConfirm = () => {
 
   const clearAmounts = useSwapperStore(state => state.clearAmounts)
   const activeSwapper = useSwapperStore(state => state.activeSwapperWithMetadata?.swapper)
+  const updateAffiliateBps = useSwapperStore(state => state.updateAffiliateBps)
+  const updateTradeAmountsFromQuote = useSwapperStore(state => state.updateTradeAmountsFromQuote)
+  const updateFees = useSwapperStore(state => state.updateFees)
+  const defaultAffiliateBps = useSwapperStore(selectSwapperDefaultAffiliateBps)
+
+  const handleDonationToggle = useCallback(async () => {
+    if (!defaultFeeAsset) {
```

Comment:
nit pick but would an early return suffice vs throwing here?

Reply:
Hmm, we'd end up with a dead click on toggling donation (instead of throwing a semi-useful error), which doesn't feel like what we want?